module/zfs/zfs_vnops.c

```diff
@@ -665,7 +665,8 @@ zfs_write(znode_t *zp, uio_t *uio, int ioflag, cred_t *cr)
 	 * If we're in replay mode, or we made no progress, return error.
 	 * Otherwise, it's at least a partial write, so it's successful.
 	 */
```

Comment:
How about:

```c
	/*
	 * If we're in replay mode, or we made no progress, or the
	 * uio data in inaccessible return an error.  Otherwise, it's
	 * at least a partial write, so it's successful.
	 */
```